core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataMappingService.java

```diff
@@ -248,7 +249,11 @@ public void refreshMapping(final String index, final String indexUUID) {
         private ClusterState applyRequest(ClusterState currentState, PutMappingClusterStateUpdateRequest request) throws IOException {
             String mappingType = request.type();
             CompressedXContent mappingUpdateSource = new CompressedXContent(request.source());
-            for (String index : request.indices()) {
+            final List<Index> indices = new ArrayList<>(request.indices().length);
+            for (String idx : request.indices()) {
+                indices.add(currentState.metaData().index(idx).getIndex());
```

Comment:
currently we ignore index names that do not exist in the [caller](https://github.com/elastic/elasticsearch/pull/17001/files#diff-d1b9f4d4f14d893cdc2a5ac6f45303aeR220) so `currentState.metaData().index(idx)` can be  null. Imo we should throw exceptions earlier and pass the list of the resolve indices to this method?